app/views/CreateChannelView.tsx

```diff
@@ -73,357 +74,295 @@ interface IOtherUser {
+				onValueChange={onValueChange}
+				testID={`create-channel-${id}`}
+				trackColor={SWITCH_TRACK_COLOR}
+				disabled={disabled}
+			/>
+		</View>
+	);
+}, shallowEqual);
+
+const CreateChannelView = () => {
+	const [channelName, setChannelName] = useState('');
+	const [type, setType] = useState(true);
+	const [readOnly, setReadOnly] = useState(false);
+	const [encrypted, setEncrypted] = useState(false);
+	const [broadcast, setBroadcast] = useState(false);
+
+	const navigation = useNavigation<StackNavigationProp<ChatsStackParamList, 'CreateChannelView'>>();
+	const { params } = useRoute<RouteProp<ChatsStackParamList, 'CreateChannelView'>>();
+	const isTeam = params?.isTeam || false;
+	const teamId = params?.teamId;
+	const { theme } = useTheme();
+	const dispatch = useDispatch();
+
+	const { encryptionEnabled, isFetching, useRealName, users } = useAppSelector(
+		state => ({
+			isFetching: state.createChannel.isFetching,
+			encryptionEnabled: state.encryption.enabled,
+			users: state.selectedUsers.users,
+			useRealName: state.settings.UI_Use_Real_Name as boolean
+		}),
+		shallowEqual
+	);
+
+	const permissions = usePermissions(['create-c', 'create-p']);
+
+	useLayoutEffect(() => {
 		navigation.setOptions({
 			title: isTeam ? I18n.t('Create_Team') : I18n.t('Create_Channel')
 		});
-	};
-
-	toggleRightButton = (channelName: string) => {
-		const { navigation } = this.props;
-		navigation.setOptions({
-			headerRight: () =>
-				channelName.trim().length > 0 && (
-					<HeaderButton.Container>
-						<HeaderButton.Item title={I18n.t('Create')} onPress={this.submit} testID='create-channel-submit' />
-					</HeaderButton.Container>
-				)
-		});
-	};
-
-	onChangeText = (channelName: string) => {
-		this.toggleRightButton(channelName);
-		this.setState({ channelName });
-	};
-
-	submit = () => {
-		const { channelName, type, readOnly, broadcast, encrypted, isTeam } = this.state;
-		const { users: usersProps, isFetching, dispatch } = this.props;
+	}, [isTeam, navigation]);
 
+	const submit = () => {
 		if (!channelName.trim() || isFetching) {
 			return;
 		}
 
 		// transform users object into array of usernames
-		const users = usersProps.map(user => user.name);
+		const usersMapped = users.map(user => user.name);
 
 		// create channel or team
 		const data = {
 			name: channelName,
-			users,
+			users: usersMapped,
 			type,
 			readOnly,
 			broadcast,
 			encrypted,
 			isTeam,
-			teamId: this.teamId
+			teamId
 		};
 		dispatch(createChannelRequest(data));
 		Review.pushPositiveEvent();
 	};
 
-	removeUser = (user: IOtherUser) => {
-		logEvent(events.CR_REMOVE_USER);
-		const { dispatch } = this.props;
-		dispatch(removeUser(user));
-	};
+	const onValueChangeType = useCallback(
+		(value: boolean) => {
+			logEvent(events.CR_TOGGLE_TYPE);
+			// If we set the channel as public, encrypted status should be false
+			setType(value);
+			setEncrypted(value && encrypted);
+		},
+		[encrypted]
+	);
+
+	const renderType = () => {
```

Comment:
This whole thing should be a different file as a `<Type />` component.
Same for the other functions.